ui/src/app/views/interactive-notebook.tsx

```diff
@@ -0,0 +1,175 @@
+  userRequestedEditMode: boolean;
+  html: string;
+}
+
+export const InteractiveNotebook = withCurrentWorkspace()(
+  class extends React.Component<Props, State> {
+
+    private billingProjectId = urlParamsStore.getValue().ns;
+    private workspaceId = urlParamsStore.getValue().wsid;
+    private nbName = urlParamsStore.getValue().nbName;
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        userRequestedEditMode: false,
+        clusterStatus: ClusterStatus.Unknown,
+        html: ''
+      };
+    }
+
+    componentDidMount(): void {
+      workspacesApi().readOnlyNotebook(this.billingProjectId, this.workspaceId, this.nbName)
+        .then(html => {
+          this.setState({html: html.html});
+        });
+    }
+
+    private runCluster(): void {
+      const retry = () => {
+        setTimeout(() => this.runCluster(), 5000);
+      };
+
+      clusterApi().listClusters(this.billingProjectId)
+        .then((body) => {
+          const cluster = body.defaultCluster;
+          this.setState({clusterStatus: cluster.status});
+
+          if (cluster.status === ClusterStatus.Stopped) {
+            notebooksClusterApi()
+              .startCluster(cluster.clusterNamespace, cluster.clusterName);
+          }
+
+          if (cluster.status === ClusterStatus.Running) {
+            this.onClusterRunning();
+          } else {
+            retry();
+          }
+        })
+        .catch(() => {
+          retry();
+        });
+    }
+
+    private onEditClick() {
+      if (this.canWrite()) {
+        this.setState({userRequestedEditMode: true});
+        this.runCluster();
+      }
+    }
+
+    private onClusterRunning() {
+      navigate(['workspaces', this.billingProjectId, this.workspaceId, 'notebooks', this.nbName]);
+    }
+
+    private canWrite() {
+      return WorkspacePermissionsUtil.canWrite(this.props.workspace.accessLevel);
+    }
+
+    render() {
+      return (
+        <div>
+          <div style={styles.navBar}>
+            <div style={{...styles.navBarItem, ...styles.active, width: 227}}>
+              Preview (Read-Only)
+            </div>
+            {this.state.userRequestedEditMode ? (
+              <div style={{...styles.navBarItem, width: 620}}>
+                <ClrIcon shape='sync' style={{...styles.navBarIcon, ...styles.rotate}}></ClrIcon>
+                Preparing your Jupyter environment. This may take up to 10 minutes.
+              </div>) : (<div>
+              <div style={
+                Object.assign({}, styles.navBarItem,
+                  this.canWrite() ? styles.clickable : styles.disabled, {width: 135})}
+                   onClick={() => { this.onEditClick(); }}>
+                <EditComponentReact enableHoverEffect={false}
+                                    disabled={!this.canWrite()}
+                                    style={{...styles.navBarIcon}}/>
+                Edit
+              </div>
+            </div>)
+            }
```

Comment:
Note: we don't have the X button. I don't think it's needed for MVP but we may want to track this.